google/cloud/bigtable/internal/defaults.cc

```diff
@@ -111,11 +109,17 @@ Options DefaultOptions(Options opts) {
   if (!opts.has<GrpcNumChannelsOption>()) {
     opts.set<GrpcNumChannelsOption>(DefaultConnectionPoolSize());
   }
-  if (!opts.has<MinConnectionRefreshOption>()) {
-    opts.set<MinConnectionRefreshOption>(kDefaultMinRefreshPeriod);
-  }
-  if (!opts.has<MaxConnectionRefreshOption>()) {
-    opts.set<MaxConnectionRefreshOption>(kDefaultMaxRefreshPeriod);
+
+  bool has_min = opts.has<MinConnectionRefreshOption>();
```

Comment:
nit: consider adding the calls to the `if()`, maybe.

Separately, consider `auto const has_min = ...`.

Reply:
Fixed. The new code is 9 lines longer (🤮), but it doesn't do any unnecessary default initializations. It is more flexible too. If we ever want to change the behavior to flip the bounds of an invalid range, there is a designated `else` block for it.